src/Net/Http/HttpOptions.php

```diff
@@ -18,6 +24,14 @@ public function __construct()
     }
 
     /**
+     * @return SslOptions
+     */
+    public function getSslOptions()
+    {
+        return $this->sslOptions ?: $this->sslOptions = new SslOptions;
```

Comment:
Actually I didn't notice the assignment, still a little overpowered getter, I still suggest another name